modin/numpy/arr.py

```diff
@@ -0,0 +1,1488 @@
+
+        Returns
+        -------
+        BasePandasDataset
+            New BasePandasDataset containing bitwise inverse to each value.
+        """
+        if not is_numeric_dtype(self.dtype):
+            raise TypeError(f"bad operand type for unary ~: '{self.dtype}'")
+        return array(_query_compiler=self._query_compiler.invert(), _ndim=self._ndim)
+
+    def _binary_op(self, other):
+        other = try_convert_from_interoperable_type(other)
+        if not isinstance(other, array):
+            raise TypeError(
+                f"Unsupported operand type(s): '{type(self)}' and '{type(other)}'"
+            )
+        broadcast = self._ndim != other._ndim
+        if broadcast:
+            # In this case, we have a 1D object doing a binary op with a 2D object
+            caller, callee = (self, other) if self._ndim == 2 else (other, self)
+            if callee.shape[0] != caller.shape[1]:
+                raise ValueError(
+                    f"operands could not be broadcast together with shapes {self.shape} {other.shape}"
+                )
+            return (caller, callee, caller._ndim, {"broadcast": broadcast, "axis": 1})
+        else:
+            if self.shape != other.shape:
+                # In this case, we either have two mismatched objects trying to do an operation
+                # or a nested 1D object that must be broadcasted trying to do an operation.
+                if self.shape[0] == other.shape[0]:
+                    matched_dimension = 0
+                elif self.shape[1] == other.shape[1]:
+                    matched_dimension = 1
+                else:
+                    raise ValueError(
+                        f"operands could not be broadcast together with shapes {self.shape} {other.shape}"
+                    )
+                if (
+                    self.shape[matched_dimension ^ 1] == 1
+                    or other.shape[matched_dimension ^ 1] == 1
+                ):
+                    return (self, other, self._ndim, {"broadcast": True, "axis": 1})
+                else:
+                    raise ValueError(
+                        f"operands could not be broadcast together with shapes {self.shape} {other.shape}"
+                    )
+            else:
+                return (self, other, self._ndim, {"broadcast": False})
+
+    def __ge__(self, x2):
+        if is_scalar(x2):
+            return array(_query_compiler=self._query_compiler.ge(x2), _ndim=self._ndim)
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        if caller._query_compiler != self._query_compiler:
+            # In this case, we are doing an operation that looks like this 1D_object >= 2D_object.
+            # For Modin to broadcast directly, we have to swap it so that the operation is actually
+            # 2D_object <= 1D_object.
+            result = caller._query_compiler.le(callee._query_compiler, **kwargs)
```

Comment:
Is `_binary_op` implicitly swapping the order of `caller`/`callee` as needed in this broadcast case? If so, is there a way to make that behavior more explicit, since it's somewhat confusing otherwise?

Reply:
It is swapping. Happy to make more explicit, but not sure how to do so off the top of my head!

Reply:
The first thing I can think of is doing something similar to the query compiler binary op, where we pass in the method and make the call something like `self._binary_op(x2, ge, le)` (i.e. pass in the operation and its inverse, and let the _binary_op method internally handle the switching of caller/callee). That's still somewhat inelegant though.